relay-server/src/actors/upstream.rs

```diff
@@ -956,6 +957,21 @@ impl Handler<IsAuthenticated> for UpstreamRelay {
     }
 }
 
+pub struct IsNetworkOutage;
+
+impl Message for IsNetworkOutage {
+    type Result = bool;
+}
+
+impl Handler<IsNetworkOutage> for UpstreamRelay {
+    type Result = bool;
+
+    fn handle(&mut self, _msg: IsNetworkOutage, _ctx: &mut Self::Context) -> Self::Result {
+        metric!(gauge(RelayGauges::NetworkOutage) = if self.is_network_outage() { 1 } else { 0 });
```

Comment:
I'm curious about the placement of the metric. 

- If we only want to emit the metric as part of the healthcheck endpoint, then let's rather move the code for that there to clearly show the intention.
- If we want to emit the metric (essentially update the gauge) every time we change the outage state, then please move the code to where the state is flipped.

Reply:
I moved the code to emit the metric to the healthcheck actor.